LuatixHQ/opencti
# [frontend/backend] Public dashboard chunk 2 (#4903)

### Proposed changes

Context recap: possibility to publish a version of a dashboard that is visible by every having the url. No need to be connected to OpenCTI. In the previous chunk (merged) we defined the process to create public dashboard and implemented one widget to feature the scenario. The objective of this PR is to generalize to all the widgets we can have in a dashboard.

* [front] refactor all existing widgets (used in private dashboard) to avoid duplicating code and be able to generate easier their public versions used in public dashboard.
* [back] create the different APIs to be able to retrieve the data for widgets.
* [front] use those new APIs in public widgets called inside public dashboards.
* [back] some refactoring on private APIs where typescript errors were found.
* [back] deleting a private dashboard also delete its public dashboards.
* [back] tests CRUD public dashboard + basic usage of widgets.

### Related issues

* #4903 

### Checklist


### Further comments

80% of the lines are refactoring and new widgets on frontend side. And there is a lot, it's not possible to read all the changes so we invite to focus more on the backend code and manual testing for the widgets.

How to test?

- Create a dashboard and add it all sort of widgets for both entities and knowledge graph,
- Click on publish button (top right) to create some public dashboards,
- Play with markings and keep an eye on the difference of data displayed in the widgets,
- Delete some public dashboards, the URL should redirect to home or login page,
- Delete the private dashboard, all associated public URL should redirect to home or login page.
